src_c/math.c

```diff
@@ -4047,6 +4047,40 @@ vector_elementwise(pgVector *vec, PyObject *_null)
     return (PyObject *)proxy;
 }
 
+static PyObject *
+math_clamp(PyObject *self, PyObject *const *args, Py_ssize_t nargs)
+{
+    const char *err_msg = "clamp requires 3 float arguments";
+
+    if (nargs != 3)
+        return RAISE(PyExc_ValueError, err_msg);
+
+    double value = PyFloat_AsDouble(args[0]);
+    if (PyErr_Occurred())
+        return RAISE(PyExc_TypeError, err_msg);
```

Comment:
By setting a new error here, the error already set by `PyFloat_AsDouble` will be discarded. So ideally this should be a `return NULL;` here (which essentially unconditionally forwards the error message)

Even in the case where it is desirable to set a custom error, you must check the older error and forward non-`TypeError`s

Reply:
I see what you mean. If PyErr_Occured, python has already set an error. Sorry, I didn't actually know anything about the python C api before contributing this, I just figured it out by reading the pygame c files. I'll spend some time to learn it properly if I ever make more complicated contributions. Anyway, I made the change, can you check to make sure I got it right?

Reply:
Yes, you got it right now!

Reply:
Great! By the way, it still says "changes requested", but I'm pretty sure I've done them all.